### ISSUE TITLE
Unable to detrmine kernel version from /boot/vmlinuz.old (Ubuntu)

### ISSUE THREAD

**Issue by sxc731** [zbm-dev/zfsbootmenu#128]
When deploying new kernel versions, Ubuntu typically provides softlinks `vmlinuz` and `vmlinuz.old` that point to the current and previous kernels, as in:
```
# ll /boot/vmlinuz*
lrwxrwxrwx 1 root root       24 Jan  6 15:34 /boot/vmlinuz -> vmlinuz-5.4.0-59-generic
-rw------- 1 root root 11682560 Dec  9 08:30 /boot/vmlinuz-5.4.0-58-generic
-rw------- 1 root root 11686656 Dec 10 12:40 /boot/vmlinuz-5.4.0-59-generic
lrwxrwxrwx 1 root root       24 Jan  6 15:34 /boot/vmlinuz.old -> vmlinuz-5.4.0-58-generic
```
This causes an issue when running `generate-zbm` without args, where it attempts to determine the latest kernel version.  The  error message is as per the title of this ticket.

The fix is simple enough; exclude soft-links by inserting the following [at line 532](https://github.com/zbm-dev/zfsbootmenu/blob/798fce838986cf1505cf46c89466e7580f7f7d12/bin/generate-zbm#L532) of `generate_zbm`;
```
next if (-l $_);
```
This works Ubuntu; not sure if there are cases where soft-links should _not_ be ignored in other distros?

**Comment by ahesford** [comment]
Can you try setting
```yaml
Kernel:
  Version: '%{current}'
```
in your `/etc/zfsbootmenu/config.yaml` and see if this fixes the issue?

**Comment by sxc731** [comment]
This does workaround the issue.  However, as a side-effect, if I run `generate_zbm` after having installed a new kernel but before rebooting, it uses the current one rather than the latest one.  I suppose it's consistent with the semantics of the config.

**Comment by zdykstra** [comment]
> This does work around the issue. However, as a side-effect, if I run `generate_zbm` after having installed a new kernel but before rebooting, it uses the current one rather than the latest one. I suppose it's consistent with the semantics of the config.

There's no need to recreate the ZBM initramfs so frequently. The kernel running in the ZBM initramfs can vary wildly from the version you `kexec` into.

**Comment by ahesford** [comment]
The right fix here is to only look for kernels of the form `<prefix>-<version>` when no version and no kernel are specified, since we expect to be able to parse the file name to extract `<prefix>-<version>` anyway. Commit ab95346 seems to do the right thing in my testing when I create a `/boot/vmlinuz.old` symlink.

If you happen to test that change on your system and notice another problem, please comment here or open a new issue.

**Comment by sxc731** [comment]
Fix tested OK on my Ubuntu 20.04 deployment, thank you.
